src/parser/mod.rs

```diff
@@ -5723,16 +5734,19 @@ impl<'a> Parser<'a> {
 
         let strict = self.parse_keyword(Keyword::STRICT);
 
-        let comment = if self.parse_keyword(Keyword::COMMENT) {
-            let _ = self.consume_token(&Token::Eq);
-            let next_token = self.next_token();
-            match next_token.token {
-                Token::SingleQuotedString(str) => Some(CommentDef::WithoutEq(str)),
-                _ => self.expected("comment", next_token)?,
-            }
-        } else {
-            None
-        };
+        // For Hive, the comment is after the table column definitions
+        if !dialect_of!(self is HiveDialect) {
+            comment = if self.parse_keyword(Keyword::COMMENT) {
+                let _ = self.consume_token(&Token::Eq);
+                let next_token = self.next_token();
+                match next_token.token {
+                    Token::SingleQuotedString(str) => Some(CommentDef::WithoutEq(str)),
```

Comment:
> For Hive, the comment is after the table column definitions

Not sure I followed this part, the comment seems to suggest this should also use `CommentDef::AfterColumnDefsWithoutEq`?

Reply:
It should be correct to use `CommentDef::WithoutEq` since it will parse comment line for other dialects except here.

Reply:
Yeah that sounds good! this was due to my misreading the dialect check